<!DOCTYPE html>
<html>
<head>
  <title>CRAN Task View: Agricultural Science</title>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8" />
  <meta name="citation_title" content="CRAN Task View: Agricultural Science" />
  <meta name="citation_author" content="Julia Piaskowski, Adam Sparks, Janet Williams" />
  <meta name="citation_publication_date" content="2022-08-08" />
  <meta name="DC.title" content="CRAN Task View: Agricultural Science" />
  <meta name="DC.creator" content="Julia Piaskowski, Adam Sparks, Janet Williams" />
  <meta name="DC.issued" content="2022-08-08" />
</head>

<body>
  <h2>CRAN Task View: Agricultural Science</h2>
  <table>
    <tr><td valign="top"><b>Maintainer:</b></td><td>Julia Piaskowski, Adam Sparks, Janet Williams</td></tr>
    <tr><td valign="top"><b>Contact:</b></td><td>&#x6a;&#x75;&#x6c;&#x69;&#x61;&#x2e;&#x70;&#x69;&#x61;&#x73;&#x6b;&#x20;&#x61;&#x74;&#x20;&#x67;&#x6d;&#x61;&#x69;&#x6c;&#x2e;&#x63;&#x6f;&#x6d;</td></tr>
    <tr><td valign="top"><b>Version:</b></td><td>2022-08-08</td></tr>
    <tr><td valign="top"><b>Source:</b></td><td><a href="https://github.com/cran-task-views/Agriculture">https://github.com/cran-task-views/Agriculture</a></td></tr>
    <tr><td valign="top"><b>Citation:</b></td><td>Julia Piaskowski, Adam Sparks, Janet Williams (2022). CRAN Task View: Agricultural Science. Version 2022-08-08.</td></tr>
  </table>

<p>Agriculture encompasses a broad breadth of disciplines. Many package in base R and contributed packages are useful to agricultural researchers. For that reason, this is not an exhaustive list of all packages useful to agricultural research. This CRAN task view is intended to cover major packages that in most cases, have been developed to support agricultural research and analytical needs.</p>
<p>Note that some of these packages are on CRAN and others are on GitHub, Bioconductor, or R-forge.</p>
<p>If you think that a package is missing from this list, please let us know by <a href="https://github.com/jpiaskowski/ctv-agriculture/issues/new?template=package_suggestion.md">filing an issue</a> in the <a href="https://github.com/IdahoAgStats/ctv-agriculture">GitHub repository</a> (preferred) or by contacting the package maintainers.</p>
<h2 id="packages-with-general-applications">Packages with general applications</h2>
<h3 id="agricultural-land-use-databases">Agricultural &amp; land use databases</h3>
<ul>
<li><p><strong>USDA databases</strong>: Data from the United States Department of Agriculture’s National Agricultural Statistical Service ‘<a href="https://www.nass.usda.gov/Quick_Stats/">Quick Stats</a>’ web API can be accessed with <a href="https://CRAN.R-project.org/package=rnassqs">rnassqs</a> or with <a href="https://CRAN.R-project.org/package=tidyUSDA">tidyUSDA</a>, which also offers some mapping capabilities. The USDA’s <a href="https://nassgeodata.gmu.edu/CropScape/">Cropland Data Layer</a> API can be accessed with <a href="https://CRAN.R-project.org/package=CropScapeR">CropScapeR</a> and <a href="https://CRAN.R-project.org/package=cdlTools">cdlTools</a>, the latter providing utility functions for processing CDL data. <a href="https://CRAN.R-project.org/package=rusda">rusda</a> provides an interface to access the USDA-ARS Systematic Mycology and Microbiology Laboratory (SMML)’s <a href="https://nt.ars-grin.gov/fungaldatabases/index.cfm">four databases</a>: Fungus-Host Distributions, Specimens, Literature and the Nomenclature database. The USDA’s Agricultural Resource Management Survey (<a href="https://www.nass.usda.gov/Surveys/Guide_to_NASS_Surveys/Ag_Resource_Management/">ARMS</a>) data API can be accessed with <a href="https://CRAN.R-project.org/package=rarms">rarms</a>. The USDA’s Livestock Mandatory Reporting data API can be accessed with <a href="https://CRAN.R-project.org/package=usdampr">usdampr</a>. The packages <a href="https://CRAN.R-project.org/package=FAOSTAT">FAOSTAT</a> and <a href="https://github.com/muuankarski/faobulk">faobulk</a> can be used to access data from the <a href="https://www.fao.org/faostat/en/#home">FAOSTAT Database</a> from the United Nations Food and Agricultural Organization (FAO).</p></li>
<li><p>Most USDA-NRCS soils related databases and APIs can be accessed with <a href="https://CRAN.R-project.org/package=soilDB">soilDB</a>.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=FedData">FedData</a> provides access to geospatial data from the United States Soil Survey Geographic (<a href="https://www.nrcs.usda.gov/wps/portal/nrcs/detail/soils/survey/?cid=nrcs142p2_053627">SSURGO</a>) database, the Global Historical Climatology Network (<a href="https://www.ncei.noaa.gov/products/land-based-station/global-historical-climatology-network-daily">GHCN</a>), the <a href="https://daymet.ornl.gov/">Daymet</a> gridded estimates of daily weather parameters for North America, the <a href="https://catalog.data.gov/dataset/international-tree-ring-data-bank-itrdb">International Tree Ring Data Bank</a>, and the <a href="https://www.usgs.gov/centers/eros/science/national-land-cover-database#overview">National Land Cover Database</a>. SSURGO data can also be accessed and processed with <a href="https://github.com/lhmrosso/XPolaris">XPolaris</a>.</p></li>
<li><p>NASA soil moisture active-passive (SMAP) data can be accessed and processed with <a href="https://CRAN.R-project.org/package=smapr">smapr</a>.</p></li>
<li><p><a href="https://github.com/INTA-Suelos/SISINTAR">SISINTAR</a> provides access to <a href="http://sisinta.inta.gob.ar/">SiSINTA</a> (Sistema de información de Suelos del INTA), a soil profile database for Argentina, and functions for processing the data.</p></li>
<li><p>SILO weather data from the <a href="https://www.longpaddock.qld.gov.au">Queensland DES Longpaddock</a> website can be accessed with <a href="https://CRAN.R-project.org/package=cropgrowdays">cropgrowdays</a>.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=febr">febr</a> has utilities to access and process data from the <a href="https://www.pedometria.org/febr">Brazilian Soil Data Repository</a>.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=PGRdup">PGRdup</a> provides functions to aid the identification of probable/possible duplicates in plant genetic resources collections.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=rfieldclimate">rfieldclimate</a> provides functionality and parsers to interact with the <a href="https://api.fieldclimate.com/v2/docs/">FieldClimate API</a>.</p></li>
</ul>
<h3 id="agricultural-data-sets">Agricultural data sets</h3>
<p>Many of the agriculture-focused packages listed in this guide also include data sets to illustrate their functionality (e.g. <a href="https://CRAN.R-project.org/package=agricolae">agricolae</a>, <a href="https://CRAN.R-project.org/package=AgroTech">AgroTech</a>, <a href="https://CRAN.R-project.org/package=BGLR">BGLR</a>).</p>
<ul>
<li><p><a href="https://CRAN.R-project.org/package=Agridat">Agridat</a> consists of a very large collection of agricultural data sets and example analyses; the package contains a vignette detailing additional data sets and extensive resources to support agricultural analysis.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=agritutorial">agritutorial</a> provides a collection of agricultural data sets and analysis with particular attention to crop experiments.</p></li>
<li><p>The <a href="https://pubmed.ncbi.nlm.nih.gov/28724064/">soybean nested associated mapping population data set</a> can be accessed via <a href="https://CRAN.R-project.org/package=SoyNAM">SoyNAM</a>.</p></li>
<li><p>The FAOSTAT data set collection for the Food and Agriculture Biomass Input–Output model (<a href="https://pubs.acs.org/doi/10.1021/acs.est.9b03554">FABIO</a>) is available through <a href="https://github.com/fineprint-global/fabio">fabio</a>.</p></li>
<li><p><a href="https://github.com/samuelbfernandes/simplePhenotypes">simplePhenotypes</a> can be used for simulating pleiotropic, linked and epistatic phenotypes.</p></li>
<li><p>USGS county data on fertilizer sales can be accessed with <a href="https://github.com/wenlong-liu/ggfertilizer">ggfertilizer</a>.</p></li>
<li><p>Annual agriculture production data from the Peruvian Integrated System of Agricultural Statistics (SIEA) covering 2004 to 2014 can be accessed with <a href="https://CRAN.R-project.org/package=cropdatape">cropdatape</a>.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=ZeBook">ZeBook</a> provides data sets and examples accompanying the book <a href="https://www.elsevier.com/books/working-with-dynamic-crop-models/wallach/978-0-12-811756-9">Working with Dynamic Crop Models</a>.</p></li>
</ul>
<h3 id="general-analytical-packages-supporting-agricultural-research">General analytical packages supporting agricultural research</h3>
<ul>
<li><p>The packages <a href="https://CRAN.R-project.org/package=nlraa">nlraa</a> and <a href="https://CRAN.R-project.org/package=AgroReg">AgroReg</a> provides linear and nonlinear regression functions specifically for agricultural applications. <a href="https://CRAN.R-project.org/package=biotools">biotools</a> can conduct a wide array of multivariate analysis for agronomists including genetic covariance, optimal plot size, tests for spatial dependence, and tests for seed lot heterogeneity.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=agriCensData">agriCensData</a> is a flexible package for working with censored data (e.g. time to flowering, instrumentation values below the detection limit, disease scoring).</p></li>
<li><p><a href="https://CRAN.R-project.org/package=grapesAgri1">grapesAgri1</a> houses a collection of shiny apps, <a href="https://www.kaugrapes.com/">GRAPES</a> (General R-shiny based Analysis Platform Empowered by Statistics), that works as a graphical user interface for individuals to upload data files and analyse. Linear models, ANOVA for CRD and 2-way RCBD designs, correlation analysis, exploratory data analysis and other common hypothesis tests are supported.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=ALUES">ALUES</a> implements methodology developed by the FAO and the International Rice Research Institute for evaluating land suitability for different crop production. f</p></li>
<li><p><a href="https://CRAN.R-project.org/package=AGPRIS">AGPRIS</a> (AGricultural PRoductivity in Space) provides functions for different spatial analyses in implemented in <a href="https://github.com/inbo/INLA">INLA</a> and other spatial approaches. The package <a href="https://CRAN.R-project.org/package=KenSyn">KenSyn</a> has example data sets and analytical code supporting the book <a href="https://www.quae.com/produit/1514/9782759228164/de-l-analyse-des-reseaux-experimentaux-a-la-meta-analyse"><strong>De L’analyse des Réseaux Expérimentaux à la Méta-analyse</strong></a> (French) or <a href="https://link.springer.com/book/10.1007/978-94-024-1696-1"><strong>From Experimental Network to Meta-analysis</strong></a> (English).</p></li>
<li><p><a href="https://CRAN.R-project.org/package=Agrotech">Agrotech</a> provides functionality for making chemical application calculations and example data sets.</p></li>
</ul>
<h2 id="discipline-specific-packages">Discipline-specific packages</h2>
<h3 id="agricultural-economics">Agricultural economics</h3>
<p>The task views for <a href="https://CRAN.R-project.org/view=Econometrics">Econometrics</a>, (Empirical) <a href="https://CRAN.R-project.org/view=Finance">Finance</a>, and <a href="https://CRAN.R-project.org/view=TimeSeries">TimeSeries</a> provide information on packages and tools relevant to agriculture economics.</p>
<ul>
<li><strong>Agricultural price forecasting</strong>: <a href="https://github.com/cran/vmdTDNN">vmdTDNN</a> forecasts univariate time series data using variational mode decomposition based time delay neural network models as described by <a href="https://doi.org/10.1109/TSP.2013.2288675">Dragomiretskiy 2014</a>. <a href="https://CRAN.R-project.org/package=stlELM">stlELM</a> also conducts univariate time series forecasting univariate time series, using seasonal-trend decomposition procedures based on loess (STL) combined with the extreme learning machine developed by <a href="https://doi.org/10.1016%2Fj.neucom.2017.11.053">Xiong 2018</a>. The package <a href="https://CRAN.R-project.org/package=eemdTDNN">eemdTDNN</a> also conduct univariate forecasting, utilizing different decomposition based time delay neural network models based on <a href="https://doi.org/10.1016%2Fj.eneco.2008.05.003">Yu 2008</a>.</li>
</ul>
<h3 id="agrometeorology">Agrometeorology</h3>
<p>The <a href="https://CRAN.R-project.org/view=Hydrology">Hydrology</a> has many resources for accessing and processing weather and climate data.</p>
<ul>
<li><p><strong>Data sources</strong>: Data from the <a href="https://cds.climate.copernicus.eu/cdsapp#!/dataset/sis-agrometeorological-indicators?tab=overview">Copernicus</a> data set of agrometeorological indicators can be downloaded and extracted using <a href="https://CRAN.R-project.org/package=ag5Tools">ag5Tools</a>. Climate crop zones in Brazil can be accessed and calculated with <a href="https://CRAN.R-project.org/package=cropZoning">cropZoning</a> using data sets from <a href="http://www.climatologylab.org/terraclimate.html">TerraClimate</a> that are calibrated to weather stations run by the National Meteorological Institute of Brazil. <a href="https://CRAN.R-project.org/package=acdcR">acdcR</a> (AgroClimatic Data by County) provides functions to calculate United States county-level variables in agricultural production or agroclimatic and weather analyses.</p></li>
<li><p><strong>Data preparation</strong>: <a href="https://CRAN.R-project.org/package=Meteor">Meteor</a> provides a set of functions for weather and climate data manipulation to support crop and crop disease modeling. <a href="https://CRAN.R-project.org/package=cropgrowdays">cropgrowdays</a> can be used for calculating growing degree days, cumulative rainfall, number of stress day, mean radiation, evapotranspiration and other variables. <a href="https://CRAN.R-project.org/package=agroclim">agroclim</a> contains functions to compute agroclimatic indices useful to zoning areas based on climatic variables and to evaluate the importance of temperature and precipitation for individual crops or in general for agricultural lands.</p></li>
<li><p>The <a href="https://github.com/anadiedrichs/frost">frost</a> package contains a compilation of empirical methods used by farmers and agronomic engineers to predict the minimum temperature to detect a frost event.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=LWFBrook90R">LWFBrook90R</a> provides an implementation of the soil vegetation atmosphere transport (SVAT) model LWF-BROOK90 to calculate daily evaporation (transpiration, interception, and soil evaporation) and soil water fluxes, along with soil water contents and soil water tension of a soil profile covered with vegetation.</p></li>
<li><p>Leaf area index and soil moisture from microwave backscattering data based on the <a href="doi:10.1016/0034-4257(93)90053-Z">WCM model</a> can be calculated with the <a href="https://CRAN.R-project.org/package=WCM">WCM</a> package.</p></li>
</ul>
<h3 id="agronomic-trials">Agronomic trials</h3>
<h4 id="experimental-design">Experimental design</h4>
<p>The task view for <a href="https://CRAN.R-project.org/view=ExperimentalDesign">ExperimentalDesign</a> provide additional information on experimental design for a wide variety of research problems.</p>
<ul>
<li><p><a href="https://CRAN.R-project.org/package=agricolae">agricolae</a> provides extensive resources for the planning and analysis of planned field experiments. Designs constructed by <a href="https://CRAN.R-project.org/package=agricolae">agricolae</a> can be visualised with <a href="https://CRAN.R-project.org/package=agricolaeplotr">agricolaeplotr</a>. Agricultural field trials layout can be also be visualised with <a href="https://CRAN.R-project.org/package=desplot">desplot</a>.</p></li>
<li><p>The package <a href="http://www.nswdpibiom.org/austatgen/software/">DiGGer</a> was developed for rectangular field trials; its purpose is to help users determine the optimal experimental design based on the treatment structure and number of replicates.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=inti">inti</a> provides functionality for experimental design and manipulation and it is focused on <a href="https://excellenceinbreeding.org/toolbox/tools/field-book">FieldBook</a> compatibility.</p></li>
</ul>
<h4 id="high-throughput-phenotyping-htp">High throughput phenotyping (HTP)</h4>
<ul>
<li><p><a href="https://CRAN.R-project.org/package=statgenHTP">statgenHTP</a> is for analyzing data from HTP platform experiments, with some functions specifically designed to work with the proprietary software <a href="https://www.vsni.co.uk/software/asreml">R-asreml</a>.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=CropDetectR">CropDetectR</a> can be used to identify crop rows from image data.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=FWRGB">FWRGB</a> can process plant images for downstream machine learning models to predict fresh biomass.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=pliman">pliman</a> provides tools for image manipulation to quantify plant leaf area, disease severity, number of disease lesions, and obtain statistics of image objects such as grains, pods, pollen, leaves, and more.</p></li>
</ul>
<h4 id="trial-analysis">Trial analysis</h4>
<ul>
<li><p><strong>General analysis</strong>: The package <a href="https://CRAN.R-project.org/package=agricolae">agricolae</a> contains functions for analyzing many common designs in agriculture trials such as split plot, lattice, Latin square and some additional functions such AMMI and AUDPC calculations. The proprietary software <a href="https://www.vsni.co.uk/software/asreml">asreml</a> provides an R version of their mixed model software for field trial analysis (note this is not open source and requires an annual license). CRAN also contains an add-on package <a href="https://CRAN.R-project.org/package=asremlPlus">asremlPlus</a> that provides several accessory functions to asreml. <a href="https://github.com/inbo/INLA">INLA</a> provides tools for Bayesian inference of latent Gaussian models, and it contains functions for modelling spatial variation, such as field experiments or farm locations. The <a href="https://CRAN.R-project.org/package=gosset">gosset</a> package provides the toolkit for a workflow to analyse experimental agriculture data, from data synthesis to model selection and visualisation. <a href="https://CRAN.R-project.org/package=AgroR">AgroR</a> has general functions and a <a href="https://agrorproject.shinyapps.io/agror_shiny/">shiny app</a> for analysis of common designs in agriculture: CRD, RCBD and Latin square.</p></li>
<li><p><strong>Spatial analysis</strong>: the <a href="https://CRAN.R-project.org/package=statgenSTA">statgenSTA</a> has functions for single trial analysis with and without spatial components. <a href="https://CRAN.R-project.org/package=SpATS">SpATS</a> can be used to adjust for field spatial variation using <a href="http://dx.doi.org/10.1002/bimj.202100212">p-splines</a>. A localised method of spatial adjustment for unreplicated trials, moving grid adjustment, is implemented with <a href="https://CRAN.R-project.org/package=mvngGrAd">mvngGrAd</a>.</p></li>
<li><p>Trials utilizing an incomplete block design can be analysed used <a href="https://CRAN.R-project.org/package=ispd">ispd</a>.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=ClimMobTools">ClimMobTools</a> is the API Client for the <a href="https://climmob.net/">ClimMob</a> citizen science platform in R for agronomic field trials.</p></li>
</ul>
<h3 id="animal-science">Animal science</h3>
<ul>
<li><p>The <a href="https://CRAN.R-project.org/view=Tracking">Tracking</a> has many resources for working with tracked animal data and studying animal movement.</p></li>
<li><p>The package <a href="https://CRAN.R-project.org/package=usdampr">usdampr</a> provides access to the USDA’s Livestock Mandatory Reporting API. Many of the genetic packages described in this task view can also be applied to animals.</p></li>
</ul>
<h3 id="breeding-quantitative-genetics">Breeding &amp; quantitative genetics</h3>
<p>See the R package repository <a href="https://www.bioconductor.org/">Bionconductor</a> for bioinformatics tools to support the processing of high-throughput genomic data.</p>
<ul>
<li><p><strong>General plant breeding functionality</strong>: <a href="https://github.com/reyzaguirre/st4gi">st4gi</a> and <a href="https://CRAN.R-project.org/package=variability">variability</a> provides several common utility functions for genetic improvement of crops. Also, please see the subsection on “genotype-by-environment interactions” in this task view for packages integrating environmental and genomic data in an analytical framework. <a href="https://CRAN.R-project.org/package=gpbStat">gpbStat</a> provides functions for common plant breeding analyses including line-by-tester analysis (<a href="http://repository.ias.ac.in/89299">Arunachalam 1974</a>) and diallel analysis (<a href="https://www.publish.csiro.au/bi/pdf/BI9560463">Griffing 1956</a>). <a href="https://R-Forge.R-project.org/projects/plantbreeding">plantbreeding</a> provides many convenience functions for working with populations and designs common in plant breeding including dialleles, line testers, augmented trials, the Carolina design, and more.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=lmDiallel">lmDiallel</a> provides service functions for analysing data sets obtained from diallel experiments, as described in <a href="https://doi.org/10.1007%2Fs00122-020-03716-8">Onofri 2020</a>.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=heritability">heritability</a> implements marker-based estimation of heritability when observations on genetically identical replicates are available.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=selection.index">selection.index</a> calculates a selection index using the method described by Smith (<a href="https://doi.org/10.1111%2Fj.1469-1809.1936.tb02143.x">1973</a>).</p></li>
<li><p><strong>Breeding simulations</strong> <a href="https://CRAN.R-project.org/package=AlphaSimR">AlphaSimR</a> is an implementation of the <a href="doi:10.3835/plantgenome2016.02.0013">AlphaSim algorithm</a> in R, providing functions for stochastic modelling of processes common to breeding programs such as selection and crossing. <a href="https://github.com/tpook92/MoBPS">MoBPS</a> has a suite of functions for simulating genetic gain and economic costs in a plant breeding program. <a href="https://CRAN.R-project.org/package=isqg">isqg</a> provides functions for high performance quantitative genetic simulations using a <a href="https://doi.org/10.1534%2Fg3.119.400373">bitset-based algorithm</a>.</p></li>
</ul>
<h4 id="linkage-mapping-qtl-analysis">Linkage mapping &amp; QTL analysis</h4>
<ul>
<li><p>There are several packages focused on <a href="https://bioconductor.org/packages/release/BiocViews.html#___LinkageDisequilibrium">linkage disequilibrium on Bioconductor</a>.</p></li>
<li><p>There are two notable and long-standing packages for quantitative trait loci (QTL) analysis: (1) <a href="https://CRAN.R-project.org/package=onemap">onemap</a>, providing <a href="https://gaow.github.io/genetic-analysis-software/m/mapmakerexp/">MapMaker/EXP</a>-like performance and extended functionality; and (2) <a href="https://CRAN.R-project.org/package=qtl">qtl</a> providing standard QTL mapping functionality and accessory functions for simulating crosses. <a href="https://CRAN.R-project.org/package=ASMap">ASMap</a> can conduct fast linkage mapping with the algorithm ‘MSTmap’. <a href="https://github.com/jendelman/MapRtools">MapRtools</a> is multi-purpose linkage mapping package for teaching and research.</p></li>
<li><p>For polyploids, the packages <a href="https://CRAN.R-project.org/package=mappoly">mappoly</a> and <a href="https://CRAN.R-project.org/package=polymapR">polymapR</a> can be used for linkage mapping and the packages <a href="https://CRAN.R-project.org/package=qtlpoly">qtlpoly</a> and <a href="https://CRAN.R-project.org/package=polyqtlR">polyqtlR</a> can be used for qtl estimation. <a href="https://github.com/jendelman/diaQTL">diaQTL</a> is for QTL and haplotype analysis of diallel populations (diploid and autotetraploid).</p></li>
<li><p><a href="https://CRAN.R-project.org/package=statgenMPP">statgenMPP</a> can conduct QTL mapping in multi-parent populations.</p></li>
<li><p>Linkage maps can be visualized with <a href="https://CRAN.R-project.org/package=LinkageMapView">LinkageMapView</a>.</p></li>
</ul>
<h4 id="gwas-genome-wide-association-studies">GWAS (Genome Wide Association Studies)</h4>
<ul>
<li><p>There are many <a href="https://bioconductor.org/packages/release/BiocViews.html#___GenomeWideAssociation">GWAS packages on Bioconductor</a>.</p></li>
<li><p>GWAS can be conducted using a <a href="https://www.nature.com/articles/ng.2314">stepwise mixed linear model</a> for multilocus data with <a href="https://CRAN.R-project.org/package=mlmm.gwas">mlmm.gwas</a> or <a href="https://github.com/Gregor-Mendel-Institute/MultLocMixMod">MultLocMixMod</a> (use <code>library(mlmm)</code> to load the package in R). The package <a href="https://CRAN.R-project.org/package=statgenGWAS">statgenGWAS</a> can fit GWAS models using the <a href="https://www.nature.com/articles/ng.548">EMMAX algorithm</a>.</p></li>
<li><p>GWAS models for a very large number of SNPs and/or observations can be estimated with <a href="https://CRAN.R-project.org/package=rMVP">rMVP</a> and <a href="https://github.com/deruncie/megaLMM">megaLMM</a>. Functions for conducting GWAS in autotetraploids are provided by <a href="https://github.com/jendelman/GWASpoly">GWASpoly</a>, and these functions also work in diploid species. Variable selection for ultra-large dimensional GWAS data sets can be done with <a href="https://CRAN.R-project.org/package=bravo">bravo</a>, which implements the Bayesian algorithm <a href="https://arxiv.org/abs/2006.07561">SVEN</a>, selection of variables with embedded screening.</p></li>
<li><p><a href="https://github.com/jendelman/StageWise">StageWise</a> provides functions to conduct a 2-stage GWAS when the phenotypic data are from multiple field trials.</p></li>
<li><p>For polyploids, <a href="https://github.com/jendelman/polyBreedR">polyBreedR</a> provides convenience functions to facilitate the use of genome-wide markers for breeding autotetraploid species, and its functionality also extends to diploids.</p></li>
</ul>
<h4 id="genomic-prediction">Genomic prediction</h4>
<ul>
<li><p><strong>General genomic selection packages</strong>: <a href="https://github.com/famuvie/breedR">breedR</a> is a general purpose package for performing quantitative genetic analyses. Genome feature mixed linear models using frequentist and Bayesian approaches can be implemented with <a href="https://CRAN.R-project.org/package=qgg">qgg</a>. The package <a href="https://CRAN.R-project.org/package=STGS">STGS</a> implements several genomic selection models for single traits. <a href="https://CRAN.R-project.org/package=BWGS">BWGS</a>, “Breed Wheat Genomic Selection”, provides a pipeline of functions for conducting genomic selection in hexaploid wheat.</p></li>
<li><p><strong>GBLUP</strong>: Packages supporting genetic prediction using mixed models augmented with pedigree or genetic marker data include <a href="https://CRAN.R-project.org/package=sommer">sommer</a>, <a href="https://CRAN.R-project.org/package=rrBLUP">rrBLUP</a>, <a href="https://CRAN.R-project.org/package=BGLR">BGLR</a>, <a href="https://github.com/perpdgo/lme4GS/"><code>lme4gs</code></a> (this package has special installation instructions), <a href="https://github.com/variani/lme4qtl">lme4qtl</a>, <a href="https://CRAN.R-project.org/package=pedigreemm">pedigreemm</a>, <a href="https://CRAN.R-project.org/package=qgtools">qgtools</a>, <a href="https://github.com/cheuerde/cpgen">cpgen</a>, <a href="https://CRAN.R-project.org/package=QTLrel">QTLrel</a>, and the licensed software <a href="https://www.vsni.co.uk/software/asreml">asreml</a>. Many of these packages have built-in functionality for data preparation steps including data imputation and calculation of the relationship matrices.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=GSelection">GSelection</a> implements genomic selection integrating additive and non-additive models.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=pedmod">pedmod</a> provides linear modelling functions integrating kinship for categorical traits.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=coxme">coxme</a> can ait Cox proportional hazards models containing both fixed and random effects with a kinship matrix.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=GSMX">GSMX</a>, multivariate genomic selection, estimates trait heritability and handles overfitting through cross validation.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=TSDFGS">TSDFGS</a> can estimate the optimal training population size and composition for genomic selection.</p></li>
<li><p><strong>Multiple environments and traits</strong>: <a href="https://CRAN.R-project.org/package=BGGE">BGGE</a> conducts genomic prediction for continuous variables, focused on genotype-by-environment genomic selection models following the methods of <a href="https://doi.org/10.1007%2Fs00122-013-2243-1">Jarquín 2014</a>. The package <a href="https://CRAN.R-project.org/package=BMTME">BMTME</a> builds genomic selection prediction models that an be expanded to multiple traits and environments using Bayesian models developed by Montesinos-Lopéx (<a href="https://doi.org/10.1534%2Fg3.116.032359">2016</a>, <a href="https://doi.org/10.1534%2Fg3.118.200728">2018a</a>, <a href="https://doi.org/10.2134%2Fagronj2018.06.0362">2018b</a>).</p></li>
<li><p><strong>Kinship and relatedness</strong>: <a href="https://CRAN.R-project.org/package=AGHmatrix">AGHmatrix</a> provides extensive options for calculating pedigree and genomic relationships (additive and dominance). The <a href="https://CRAN.R-project.org/package=pedigree">pedigree</a> packages provides functionality for ordering pedigrees, calculating and inverting the pedigree relationship matrix and other related tasks. <a href="https://CRAN.R-project.org/package=statgenIBD">statgenIBD</a> can calculate IBD probabilities for biparental, three-way and four-way crosses. <a href="https://CRAN.R-project.org/package=kinship2">kinship2</a> provides functions for manipulating and visualising pedigree-based kinship data.</p></li>
</ul>
<h3 id="crop-growth-models-crop-modelling">Crop growth models &amp; crop modelling</h3>
<ul>
<li><p>The <a href="https://CRAN.R-project.org/package=apsimx">apsimx</a> package has functions to read, inspect, edit and run files for APSIM “Next Generation” (<code>.json</code>, <code>.apsimx</code>) and APSIM “Classic” (<code>.xml</code>, <code>.apsim</code>) files. <a href="https://CRAN.R-project.org/package=rapsimng">rapsimng</a> works with next generation APSIM files.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=DSSAT">DSSAT</a> provides a comprehensive R interface to the Decision Support System for Agrotechnology Transfer Cropping Systems Model (DSSAT-CSM) documented by <a href="https://doi.org/10.1016/S1161-0301(02)00107-7">Jones (2003)</a>. This package provides cross-platform functions to read and write input files, run DSSAT-CSM, and read output files.</p></li>
<li><p>The modelling framework <a href="www.simplace.net">Simplace</a> (Scientific Impact assessment and Modelling Platform for Advanced Crop and Ecosystem management) can be accessed using <a href="https://github.com/gk-crop/simplace_rpkg">simplace_rpkg</a>.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=Meteor">Meteor</a> provides a set of functions for weather and climate data manipulation to support crop and crop disease modelling.</p></li>
<li><p><strong>Crop Water Usage</strong>: <a href="https://CRAN.R-project.org/package=cropDemand">cropDemand</a> can be used to estimate crop water demand in Brazilian production regions using the <a href="http://www.climatologylab.org/terraclimate.html">TerraClimate</a> data set. <a href="https://CRAN.R-project.org/package=Evapotranspiration">Evapotranspiration</a> can estimate potential and actual evapotranspiration using 21 different models.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=metrica">metrica</a> has many convenience functions for comparing model predictions with ground truth data.</p></li>
<li><p><a href="https://github.com/cropmodels/phenorice">phenorice</a> is an R implementation of the <a href="http://dx.doi.org/10.1016/j.rse.2017.03.029">phenorice model</a> for remote sensing of rice crop production. <a href="https://github.com/lbusett/phenoriceR">phenoriceR</a> provides helper functions for processing data from the phenorice model.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=Recocrop">Recocrop</a> estimates environmental suitability for plants using a limiting factor approach for plant growth following <a href="doi:10.1007/BF00045728">Hackett (1991)</a>.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=Rquefts">Rquefts</a> provides an implementation of the QUEFTS (Quantitative Evaluation of the Native Fertility of Tropical Soils) model (<a href="https://doi.org/10.1016%2F0016-7061%2890%2990021-Z">Janssen 1990</a>).</p></li>
<li><p><a href="https://CRAN.R-project.org/package=Rwofost">Rwofost</a> is an implementation of the WOFOST (“World Food Studies”) crop growth model (<a href="https://doi.org/10.1016%2Fj.agsy.2018.06.018">De Wit 2019</a>).</p></li>
</ul>
<h3 id="entomology">Entomology</h3>
<ul>
<li>The <a href="https://CRAN.R-project.org/view=Survival">Survival</a> task view lists resources for working with censored data. The package <a href="https://github.com/OnofriAndreaPG/agriCensData">agriCensData</a> provides functions for dealing with censored data in common agricultural contexts.</li>
<li><a href="https://CRAN.R-project.org/package=hnp">hnp</a> Generates half-normal plots with simulation envelopes using different diagnostics from a range of different fitted models.</li>
</ul>
<h3 id="food-science">Food science</h3>
<ul>
<li><p>For packages supporting sensory studies, see the <a href="https://CRAN.R-project.org/view=Psychometrics">Psychometrics</a> task view.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=NutrienTrackeR">NutrienTrackeR</a> provides convenience functions for calculating nutrient content (macronutrients and micronutrients) of foods using food composition data from several reference databases, including: ‘USDA’ (United States), ‘CIQUAL’ (France), ‘BEDCA’ (Spain) and ‘CNF’ (Canada).</p></li>
</ul>
<h3 id="genotype-by-environment-interactions">Genotype-by-environment interactions</h3>
<ul>
<li><p><a href="https://CRAN.R-project.org/package=statgenGxE">statgenGxE</a> implements several analytical approaches for addressing genotype-by-environment interactions.</p></li>
<li><p>The package <a href="https://CRAN.R-project.org/package=gge">gge</a> can generate GGE biplots, while <a href="https://CRAN.R-project.org/package=bayesammi">bayesammi</a> can conduct Bayesian estimation of additive main effects multiplicative interaction (AMMI) model.</p></li>
<li><p><a href="https://github.com/allogamous/EnvRtype">EnvRtype</a> can be used for assembling climate data, data set preparation and environmental classification or envirotyping.</p></li>
<li><p><a href="https://github.com/lian0090/FW">FW</a> implements Finlay-Wilkinson regression using a Gibbs sampler; <a href="https://CRAN.R-project.org/package=spFW">spFW</a> also conducts spatial Finlay-Wilkinson analysis for multi-environmental trials using a Bayesian hierarchical model.</p></li>
<li><p>A wide variety of stability analysis statistics can be calculated via <a href="https://CRAN.R-project.org/package=agrostab">agrostab</a> including coefficient of homeostaticity, specific adaptive ability, weighted homeostaticity index, <a href="https://doi.org/10.4141%2Fcjps88-018">superiority measure</a>, <a href="https://doi.org/10.2135%2Fcropsci1966.0011183X000600010011x">regression on environmental index</a>, <a href="https://doi.org/10.2135%2Fcropsci1971.0011183X001100020006x">Tai’s stability parameters</a>, <a href="https://doi.org/10.1038%2Fhdy.1972.87">stability variance</a>, ecovalence and other stability parameters.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=IBCF.MTME">IBCF.MTME</a> implements item-based collaborative filtering for continuous data in multi-trait and multi-environment trials following the methods described by <a href="https://acsess.onlinelibrary.wiley.com/doi/10.3835/plantgenome2018.02.0013">Montesinos-López (2018)</a></p></li>
</ul>
<h3 id="plant-pathology">Plant pathology</h3>
<ul>
<li><p>The <a href="https://CRAN.R-project.org/view=Epidemiology">Epidemiology</a> task view lists relevant package for modelling plant diseases.</p></li>
<li><p><strong>Epidemiology Simulation</strong>: Stochastic disease modelling of plant pathogens incorporating spatial and genetic information can be done with <a href="https://CRAN.R-project.org/package=landsepi">landsepi</a>. The package <a href="https://CRAN.R-project.org/package=ascotraceR">ascotraceR</a> can simulate an Ascochyta blight infection in a chickpea field following the model developed by <a href="https://doi.org/10.1094%2FPHYTO.2002.92.10.1110">Diggle (2022)</a>).</p></li>
<li><p><a href="https://CRAN.R-project.org/package=epiphy">epiphy</a> is a toolbox for analyzing plant disease epidemics. It provides a common framework for plant disease intensity data recorded over time and/or space.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=epifitter">epifitter</a> provides functions for analysis and visualization of plant disease progress curve data.</p></li>
<li><p><strong>Plant Pathogen Genetics</strong>: <a href="https://CRAN.R-project.org/package=hagis">hagis</a> has functions for analysis of plant pathogen pathotype survey data. Functions provided calculate distribution of susceptibilities, distribution of complexities with statistics, pathotype frequency distribution, as well as diversity indices for pathotypes.Evolution of resistance genes under pesticide pressure can be simulated under different numbers of pests, modes of pest reproduction, resistance loci, number of pesticides and other facets with <a href="https://CRAN.R-project.org/package=resevol">resevol</a>. Populations with mixed clonal/sexual reproductive strategies can be analyzed with <a href="https://CRAN.R-project.org/package=poppr">poppr</a>, which has population genetic analysis tools for hierarchical analysis of partially clonal populations.</p></li>
</ul>
<h3 id="rural-sociology">Rural sociology</h3>
<ul>
<li><p>See the task view for <a href="https://CRAN.R-project.org/view=Psychometrics">Psychometrics</a> for general sociology packages.</p></li>
<li><p>Both the <a href="https://CRAN.R-project.org/view=Survival">Survival</a> task view and the <a href="https://CRAN.R-project.org/package=agriCensData">agriCensData</a> package provide tools for working with interval and censored data.</p></li>
</ul>
<h3 id="soil-science-and-precision-agriculture">Soil science and precision agriculture</h3>
<ul>
<li><p><strong>Spatial</strong>: The <a href="https://CRAN.R-project.org/view=Spatial">Spatial</a> and <a href="https://CRAN.R-project.org/view=SpatioTemporal">SpatioTemporal</a> CRAN task views provide extensive resources in spatial statistics. <a href="https://CRAN.R-project.org/package=mpspline2">mpspline2</a> implements a mass-preserving spline to soil attributes to make continuous down-profile estimates of attributes measured over discrete, often discontinuous depth intervals.</p></li>
<li><p>The <a href="https://CRAN.R-project.org/package=sharpshootR">sharpshootR</a> contains a compendium of utility functions supporting soils survey work including data management, summary, visualisations and conversions.</p></li>
<li><p>For soil pedology, <a href="https://CRAN.R-project.org/package=aqp">aqp</a> provides a general toolkit for soil scientists: specialized data structures, soil profile summary, visualisation, color conversion, and more. <a href="https://CRAN.R-project.org/package=SoilTaxonomy">SoilTaxonomy</a> provides functions for parsing soil taxonomic terms. <a href="https://CRAN.R-project.org/package=pedometrics">pedometrics</a> has many utility functions for common analyses of soil data.</p></li>
<li><p><strong>Soil water</strong>: Soil water retention curves can be calculated by the <a href="https://CRAN.R-project.org/package=soilwater">soilwater</a> packages using the <a href="https://doi.org/10.2136/sssaj1980.03615995004400050002x">Van Genuchten (1980)</a> method for soil water retention and <a href="https://doi.org/10.1029/WR012i003p00513">Mualem (1976)</a> method for hydraulic conductivity. Estimation and prediction of parameters of soil hydraulic property models can be accomplished with <a href="https://CRAN.R-project.org/package=spsh">spsh</a>.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=SoilR">SoilR</a> models soil organic matter decomposition in terrestrial ecosystems with linear and nonlinear models. The <a href="https://CRAN.R-project.org/package=sorcering">sorcering</a> can be used to model soil organic carbon and soil organic nitrogen and to calculate N mineralisation rates.</p></li>
<li><p>Soil texture triangles can be graphed using <a href="https://CRAN.R-project.org/package=soiltexture">soiltexture</a>; this package can also classify and transform soil texture data.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=QI">QI</a> can be used to calculate potassium intensity and exchangeability.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=DMMF">DMMF</a> implements the daily based Morgan-Morgan-Finney (DMMF) soil erosion model (<a href="https://doi.org/10.3390%2Fw9040278">Choi, 2017</a>) for estimating surface runoff and sediment budgets from a field or a catchment on a daily basis.</p></li>
<li><p><strong>Soil Fertility Testing</strong>: <a href="https://CRAN.R-project.org/package=soiltestcorr">soiltestcorr</a> has functions for conducting correlation analysis between soil test values and crop yield data. <a href="https://CRAN.R-project.org/package=SoilTesting">SoilTesting</a> provides functions for calculating soil mineral concentrations from analytical lab results. <a href="https://github.com/mbask/fertplan">fertplan</a> provides fertilizer recommendations based on soil test results (note this packages is optimized for horticultural crop production in Italy).</p></li>
<li><p><strong>Remote Sensing</strong>: Agriculture image features from spectral data can extracted with <a href="https://CRAN.R-project.org/package=agrifeature">agrifeature</a>. It has functions to calculate gray level co-occurrence matrix (GLCM), RGB-based vegetative index (RGB VI) and normalized difference vegetation index (NDVI). Experimental units (e.g. plots) can be obtained from spectral images using <a href="https://CRAN.R-project.org/package=rPAex">rPAex</a>. The <a href="https://CRAN.R-project.org/package=mapsRinteractive">mapsRinteractive</a> package provides functions for working with soil point data in raster format.</p></li>
<li><p>The <a href="https://www.fao.org/land-water/databases-and-software/crop-information/en/">suitability</a> of specific soils for crop production can be analyzed using <a href="https://CRAN.R-project.org/package=soilassessment">soilassessment</a>, including soil fertility classes, soil erosion models and soil salinity classification. Suitability requirements are for crops grouped into cereal crops, nuts, legumes, fruits, vegetables, industrial crops, and root crops.</p></li>
</ul>
<h3 id="weed-science">Weed science</h3>
<p>For ecological studies and analytical applications, the <a href="https://CRAN.R-project.org/view=Environmetrics">Environmetrics</a> task view provides a list of existing R resources in this topic.</p>
<ul>
<li><p><strong>Dose Response</strong>: the package <a href="https://CRAN.R-project.org/package=drc">drc</a> offers versatile model fitting and after-fitting functions for dose-response curves. <a href="https://CRAN.R-project.org/package=LW1949">LW1949</a> implements the Litchefield and Wilcoxon (<a href="http://jpet.aspetjournals.org/content/96/2/99.abstract">1949</a>)dose-response model.</p></li>
<li><p><a href="https://CRAN.R-project.org/package=PROSPER">PROSPER</a> is a package for simulating weed population dynamics at the individual and population level under a range of conditions including herbicide resistance and herbicide pressure.</p></li>
</ul>

  <h3>CRAN packages</h3>
  <table>
    <tr valign="top"><td><i>Core:</i></td><td><a href="https://CRAN.R-project.org/package=AGHmatrix">AGHmatrix</a>, <a href="https://CRAN.R-project.org/package=agriCensData">agriCensData</a>, <a href="https://CRAN.R-project.org/package=Agridat">Agridat</a>, <a href="https://CRAN.R-project.org/package=apsimx">apsimx</a>, <a href="https://CRAN.R-project.org/package=aqp">aqp</a>, <a href="https://CRAN.R-project.org/package=cdlTools">cdlTools</a>, <a href="https://CRAN.R-project.org/package=drc">drc</a>, <a href="https://CRAN.R-project.org/package=DSSAT">DSSAT</a>, <a href="https://CRAN.R-project.org/package=FedData">FedData</a>, <a href="https://CRAN.R-project.org/package=inti">inti</a>, <a href="https://CRAN.R-project.org/package=Meteor">Meteor</a>, <a href="https://CRAN.R-project.org/package=nlraa">nlraa</a>, <a href="https://CRAN.R-project.org/package=qtl">qtl</a>, <a href="https://CRAN.R-project.org/package=sommer">sommer</a>, <a href="https://CRAN.R-project.org/package=tidyUSDA">tidyUSDA</a>.</td></tr>
    <tr valign="top"><td><i>Regular:</i></td><td><a href="https://CRAN.R-project.org/package=acdcR">acdcR</a>, <a href="https://CRAN.R-project.org/package=ag5Tools">ag5Tools</a>, <a href="https://CRAN.R-project.org/package=AGPRIS">AGPRIS</a>, <a href="https://CRAN.R-project.org/package=agricolae">agricolae</a>, <a href="https://CRAN.R-project.org/package=agricolaeplotr">agricolaeplotr</a>, <a href="https://CRAN.R-project.org/package=agrifeature">agrifeature</a>, <a href="https://CRAN.R-project.org/package=agritutorial">agritutorial</a>, <a href="https://CRAN.R-project.org/package=agroclim">agroclim</a>, <a href="https://CRAN.R-project.org/package=AgroR">AgroR</a>, <a href="https://CRAN.R-project.org/package=AgroReg">AgroReg</a>, <a href="https://CRAN.R-project.org/package=agrostab">agrostab</a>, <a href="https://CRAN.R-project.org/package=AgroTech">AgroTech</a>, <a href="https://CRAN.R-project.org/package=Agrotech">Agrotech</a>, <a href="https://CRAN.R-project.org/package=AlphaSimR">AlphaSimR</a>, <a href="https://CRAN.R-project.org/package=ALUES">ALUES</a>, <a href="https://CRAN.R-project.org/package=ascotraceR">ascotraceR</a>, <a href="https://CRAN.R-project.org/package=ASMap">ASMap</a>, <a href="https://CRAN.R-project.org/package=asremlPlus">asremlPlus</a>, <a href="https://CRAN.R-project.org/package=bayesammi">bayesammi</a>, <a href="https://CRAN.R-project.org/package=BGGE">BGGE</a>, <a href="https://CRAN.R-project.org/package=BGLR">BGLR</a>, <a href="https://CRAN.R-project.org/package=biotools">biotools</a>, <a href="https://CRAN.R-project.org/package=BMTME">BMTME</a>, <a href="https://CRAN.R-project.org/package=bravo">bravo</a>, <a href="https://CRAN.R-project.org/package=BWGS">BWGS</a>, <a href="https://CRAN.R-project.org/package=ClimMobTools">ClimMobTools</a>, <a href="https://CRAN.R-project.org/package=coxme">coxme</a>, <a href="https://CRAN.R-project.org/package=cropdatape">cropdatape</a>, <a href="https://CRAN.R-project.org/package=cropDemand">cropDemand</a>, <a href="https://CRAN.R-project.org/package=CropDetectR">CropDetectR</a>, <a href="https://CRAN.R-project.org/package=cropgrowdays">cropgrowdays</a>, <a href="https://CRAN.R-project.org/package=CropScapeR">CropScapeR</a>, <a href="https://CRAN.R-project.org/package=cropZoning">cropZoning</a>, <a href="https://CRAN.R-project.org/package=desplot">desplot</a>, <a href="https://CRAN.R-project.org/package=DMMF">DMMF</a>, <a href="https://CRAN.R-project.org/package=eemdTDNN">eemdTDNN</a>, <a href="https://CRAN.R-project.org/package=epifitter">epifitter</a>, <a href="https://CRAN.R-project.org/package=epiphy">epiphy</a>, <a href="https://CRAN.R-project.org/package=Evapotranspiration">Evapotranspiration</a>, <a href="https://CRAN.R-project.org/package=FAOSTAT">FAOSTAT</a>, <a href="https://CRAN.R-project.org/package=febr">febr</a>, <a href="https://CRAN.R-project.org/package=FWRGB">FWRGB</a>, <a href="https://CRAN.R-project.org/package=gge">gge</a>, <a href="https://CRAN.R-project.org/package=gosset">gosset</a>, <a href="https://CRAN.R-project.org/package=gpbStat">gpbStat</a>, <a href="https://CRAN.R-project.org/package=grapesAgri1">grapesAgri1</a>, <a href="https://CRAN.R-project.org/package=GSelection">GSelection</a>, <a href="https://CRAN.R-project.org/package=GSMX">GSMX</a>, <a href="https://CRAN.R-project.org/package=hagis">hagis</a>, <a href="https://CRAN.R-project.org/package=heritability">heritability</a>, <a href="https://CRAN.R-project.org/package=hnp">hnp</a>, <a href="https://CRAN.R-project.org/package=IBCF.MTME">IBCF.MTME</a>, <a href="https://CRAN.R-project.org/package=ispd">ispd</a>, <a href="https://CRAN.R-project.org/package=isqg">isqg</a>, <a href="https://CRAN.R-project.org/package=KenSyn">KenSyn</a>, <a href="https://CRAN.R-project.org/package=kinship2">kinship2</a>, <a href="https://CRAN.R-project.org/package=landsepi">landsepi</a>, <a href="https://CRAN.R-project.org/package=LinkageMapView">LinkageMapView</a>, <a href="https://CRAN.R-project.org/package=lmDiallel">lmDiallel</a>, <a href="https://CRAN.R-project.org/package=LW1949">LW1949</a>, <a href="https://CRAN.R-project.org/package=LWFBrook90R">LWFBrook90R</a>, <a href="https://CRAN.R-project.org/package=mappoly">mappoly</a>, <a href="https://CRAN.R-project.org/package=mapsRinteractive">mapsRinteractive</a>, <a href="https://CRAN.R-project.org/package=metrica">metrica</a>, <a href="https://CRAN.R-project.org/package=mlmm.gwas">mlmm.gwas</a>, <a href="https://CRAN.R-project.org/package=mpspline2">mpspline2</a>, <a href="https://CRAN.R-project.org/package=mvngGrAd">mvngGrAd</a>, <a href="https://CRAN.R-project.org/package=NutrienTrackeR">NutrienTrackeR</a>, <a href="https://CRAN.R-project.org/package=onemap">onemap</a>, <a href="https://CRAN.R-project.org/package=pedigree">pedigree</a>, <a href="https://CRAN.R-project.org/package=pedigreemm">pedigreemm</a>, <a href="https://CRAN.R-project.org/package=pedmod">pedmod</a>, <a href="https://CRAN.R-project.org/package=pedometrics">pedometrics</a>, <a href="https://CRAN.R-project.org/package=PGRdup">PGRdup</a>, <a href="https://CRAN.R-project.org/package=pliman">pliman</a>, <a href="https://CRAN.R-project.org/package=polymapR">polymapR</a>, <a href="https://CRAN.R-project.org/package=polyqtlR">polyqtlR</a>, <a href="https://CRAN.R-project.org/package=poppr">poppr</a>, <a href="https://CRAN.R-project.org/package=PROSPER">PROSPER</a>, <a href="https://CRAN.R-project.org/package=qgg">qgg</a>, <a href="https://CRAN.R-project.org/package=qgtools">qgtools</a>, <a href="https://CRAN.R-project.org/package=QI">QI</a>, <a href="https://CRAN.R-project.org/package=qtlpoly">qtlpoly</a>, <a href="https://CRAN.R-project.org/package=QTLrel">QTLrel</a>, <a href="https://CRAN.R-project.org/package=rapsimng">rapsimng</a>, <a href="https://CRAN.R-project.org/package=rarms">rarms</a>, <a href="https://CRAN.R-project.org/package=Recocrop">Recocrop</a>, <a href="https://CRAN.R-project.org/package=resevol">resevol</a>, <a href="https://CRAN.R-project.org/package=rfieldclimate">rfieldclimate</a>, <a href="https://CRAN.R-project.org/package=rMVP">rMVP</a>, <a href="https://CRAN.R-project.org/package=rnassqs">rnassqs</a>, <a href="https://CRAN.R-project.org/package=rPAex">rPAex</a>, <a href="https://CRAN.R-project.org/package=Rquefts">Rquefts</a>, <a href="https://CRAN.R-project.org/package=rrBLUP">rrBLUP</a>, <a href="https://CRAN.R-project.org/package=rusda">rusda</a>, <a href="https://CRAN.R-project.org/package=Rwofost">Rwofost</a>, <a href="https://CRAN.R-project.org/package=selection.index">selection.index</a>, <a href="https://CRAN.R-project.org/package=sharpshootR">sharpshootR</a>, <a href="https://CRAN.R-project.org/package=smapr">smapr</a>, <a href="https://CRAN.R-project.org/package=soilassessment">soilassessment</a>, <a href="https://CRAN.R-project.org/package=soilDB">soilDB</a>, <a href="https://CRAN.R-project.org/package=SoilR">SoilR</a>, <a href="https://CRAN.R-project.org/package=SoilTaxonomy">SoilTaxonomy</a>, <a href="https://CRAN.R-project.org/package=soiltestcorr">soiltestcorr</a>, <a href="https://CRAN.R-project.org/package=SoilTesting">SoilTesting</a>, <a href="https://CRAN.R-project.org/package=soiltexture">soiltexture</a>, <a href="https://CRAN.R-project.org/package=soilwater">soilwater</a>, <a href="https://CRAN.R-project.org/package=sorcering">sorcering</a>, <a href="https://CRAN.R-project.org/package=SoyNAM">SoyNAM</a>, <a href="https://CRAN.R-project.org/package=SpATS">SpATS</a>, <a href="https://CRAN.R-project.org/package=spFW">spFW</a>, <a href="https://CRAN.R-project.org/package=spsh">spsh</a>, <a href="https://CRAN.R-project.org/package=statgenGWAS">statgenGWAS</a>, <a href="https://CRAN.R-project.org/package=statgenGxE">statgenGxE</a>, <a href="https://CRAN.R-project.org/package=statgenHTP">statgenHTP</a>, <a href="https://CRAN.R-project.org/package=statgenIBD">statgenIBD</a>, <a href="https://CRAN.R-project.org/package=statgenMPP">statgenMPP</a>, <a href="https://CRAN.R-project.org/package=statgenSTA">statgenSTA</a>, <a href="https://CRAN.R-project.org/package=STGS">STGS</a>, <a href="https://CRAN.R-project.org/package=stlELM">stlELM</a>, <a href="https://CRAN.R-project.org/package=TSDFGS">TSDFGS</a>, <a href="https://CRAN.R-project.org/package=usdampr">usdampr</a>, <a href="https://CRAN.R-project.org/package=variability">variability</a>, <a href="https://CRAN.R-project.org/package=WCM">WCM</a>, <a href="https://CRAN.R-project.org/package=ZeBook">ZeBook</a>.</td></tr>
  </table>

  <h3>Related links</h3>
  <ul>
    <li>General web-based tutorials: <a href="https://rstats4ag.org/">R for Agricultural Statistics</a>, <a href="agstats.io">AgStats.io</a>, <a href="https://schmidtpaul.github.io/DSFAIR/">Data Science for Agriculture in R</a></li>
    <li>Website: <a href="https://schmidtpaul.github.io/MMFAIR/">Mixed Models in Agriculture</a></li>
    <li>Additional data sets: the Github repository <a href="https://github.com/OnofriAndreaPG/agroBioData">agroBioData</a> houses a collection of data sets supporting agriculture and applied biology (note that this is a collection of CSV files and a <code>.rda</code> file, not a package)</li>
    <li>Book: <strong><a href="Applied%20Statistics%20in%20Agricultural,%20Biological,%20and%20Environmental%20Sciences">Applied Statistics in Agricultural, Biological, and Environmental Sciences</a></strong> (2017) edited by B Glaz and KM Yeater.</li>
    <li>Books published by <a href="https://acsess.onlinelibrary.wiley.com/hub/books">ASA/CSSA/SSSA</a></li>
    <li>Books published by the <a href="https://ashs.org/store/ListProducts.aspx?catid=3379250">ASHS</a></li>
    <li>Books published by the <a href="https://wssa.net/publications/">WSSA</a></li>
    <li>Books published by the <a href="https://my.apsnet.org/APSStore/">APS</a></li>
    <li>Publications by the <a href="https://online.entsoc.org/esassa/ecssashop.shopping_page">ESA</a></li>
    <li>Teaching curriculum: <strong><a href="https://r-forge.r-project.org/projects/cropcc/">Crop Climate Change Course</a></strong>: materials for the course <em>Analyzing the Impact of Climate Change on Crops and Varieties</em></li>
  </ul>

  <h3>Other resources</h3>
  <ul>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Econometrics">Econometrics</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Environmetrics">Environmetrics</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Epidemiology">Epidemiology</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=ExperimentalDesign">ExperimentalDesign</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Finance">Finance</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Hydrology">Hydrology</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Psychometrics">Psychometrics</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Spatial">Spatial</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=SpatioTemporal">SpatioTemporal</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Survival">Survival</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=TimeSeries">TimeSeries</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Tracking">Tracking</a></li>
    <li>R-Forge Project: <a href="https://R-Forge.R-project.org/projects/plantbreeding">plantbreeding</a></li>
    <li>GitHub Project: <a href="https://github.com/OnofriAndreaPG/agriCensData">agriCensData</a></li>
    <li>GitHub Project: <a href="https://github.com/famuvie/breedR">breedR</a></li>
    <li>GitHub Project: <a href="https://github.com/cheuerde/cpgen">cpgen</a></li>
    <li>GitHub Project: <a href="https://github.com/jendelman/diaQTL">diaQTL</a></li>
    <li>GitHub Project: <a href="https://github.com/allogamous/EnvRtype">EnvRtype</a></li>
    <li>GitHub Project: <a href="https://github.com/fineprint-global/fabio">fabio</a></li>
    <li>GitHub Project: <a href="https://github.com/muuankarski/faobulk">faobulk</a></li>
    <li>GitHub Project: <a href="https://github.com/mbask/fertplan">fertplan</a></li>
    <li>GitHub Project: <a href="https://github.com/anadiedrichs/frost">frost</a></li>
    <li>GitHub Project: <a href="https://github.com/lian0090/FW">FW</a></li>
    <li>GitHub Project: <a href="https://github.com/wenlong-liu/ggfertilizer">ggfertilizer</a></li>
    <li>GitHub Project: <a href="https://github.com/jendelman/GWASpoly">GWASpoly</a></li>
    <li>GitHub Project: <a href="https://github.com/inbo/INLA">INLA</a></li>
    <li>GitHub Project: <a href="https://github.com/variani/lme4qtl">lme4qtl</a></li>
    <li>GitHub Project: <a href="https://github.com/jendelman/MapRtools">MapRtools</a></li>
    <li>GitHub Project: <a href="https://github.com/deruncie/megaLMM">megaLMM</a></li>
    <li>GitHub Project: <a href="https://github.com/tpook92/MoBPS">MoBPS</a></li>
    <li>GitHub Project: <a href="https://github.com/Gregor-Mendel-Institute/MultLocMixMod">MultLocMixMod</a></li>
    <li>GitHub Project: <a href="https://github.com/cropmodels/phenorice">phenorice</a></li>
    <li>GitHub Project: <a href="https://github.com/lbusett/phenoriceR">phenoriceR</a></li>
    <li>GitHub Project: <a href="https://github.com/jendelman/polyBreedR">polyBreedR</a></li>
    <li>GitHub Project: <a href="https://github.com/gk-crop/simplace_rpkg">simplace_rpkg</a></li>
    <li>GitHub Project: <a href="https://github.com/samuelbfernandes/simplePhenotypes">simplePhenotypes</a></li>
    <li>GitHub Project: <a href="https://github.com/INTA-Suelos/SISINTAR">SISINTAR</a></li>
    <li>GitHub Project: <a href="https://github.com/reyzaguirre/st4gi">st4gi</a></li>
    <li>GitHub Project: <a href="https://github.com/jendelman/StageWise">StageWise</a></li>
    <li>GitHub Project: <a href="https://github.com/cran/vmdTDNN">vmdTDNN</a></li>
    <li>GitHub Project: <a href="https://github.com/lhmrosso/XPolaris">XPolaris</a></li>
  </ul>

</body>
</html>
